crates/core/src/ring_switch/prove.rs

```diff
@@ -136,18 +142,30 @@ where
 			|PIOPSumcheckClaimDesc {
 			     committed_idx,
 			     suffix_desc_idx,
-			     eval_claim: _,
+			     eval_claim,
 			 }| {
-				let suffix = &system.suffix_descs[*suffix_desc_idx];
-				let suffix_query = &suffix_queries[*suffix_desc_idx];
-				let partial_eval =
-					witnesses[*committed_idx].evaluate_partial_high(suffix_query.to_ref())?;
-				TowerTensorAlgebra::new(
-					suffix.kappa,
+				let suffix_desc = &system.suffix_descs[*suffix_desc_idx];
+
+				let elems = if let Some(partial_eval) =
+					memoized_data.partial_eval(eval_claim.id, Arc::as_ref(&suffix_desc.suffix))
+				{
+					PackedField::iter_slice(
```

Comment:
Deduplicate this between branches. Consider an interface for MemoizedData that [tells instead of asking](https://martinfowler.com/bliki/TellDontAsk.html) (like `MemoizedData::full_query`).

Reply:
Not sure that I can apply it here, in the first case we look up partial_eval by ID, and in the second case it's a evaluate_partial_high MultilinearPoly.
Yes, we could do something like `partial_eval(witness: M, witness_id: OracleId, eval_point: &[F])`, but that seems a bit overloaded.